core/circuit_breaker/rule_manager.go

```diff
@@ -82,15 +173,25 @@ func onRuleUpdate(rules []Rule) (err error) {
 	for k, v := range newBreakerRules {
 		cbs := make([]CircuitBreaker, 0, len(v))
 		for _, r := range v {
-			cbs = append(cbs, r.convert2CircuitBreaker())
+			generator := cbGenFuncMap[r.BreakerStrategy()]
+			if generator == nil {
+				logger.Warnf("Circuit Breaker Generator for %+v is not existed.", r.BreakerStrategy())
+				continue
+			}
+			cb := generator(r)
```

Comment:
Here we may improve the logic: if a rule remains unchanged (identical), use the former circuit breaker instance instead of re-creating a new one.